src/frontend/flight/handler.rs

```diff
@@ -186,14 +192,49 @@ impl SeafowlFlightHandler {
                             .await?
                     }
                 };
-                let url = log_store.root_uri();
-                (LakehouseSyncTarget::Delta(log_store), url)
+                LakehouseSyncTarget::Delta(log_store)
             }
             TableFormat::Iceberg => {
-                return Err(SyncError::NotImplemented);
+                let (location, file_io) = match cmd.store {
+                    None => {
+                        return Err(SyncError::NotImplemented);
+                    }
+                    Some(store_loc) => {
+                        let location = store_loc.location;
+                        let options = store_loc.options;
+                        let file_io_props = object_store_opts_to_file_io_props(&options);
+                        let file_io = FileIO::from_path(&location)
+                            .unwrap()
+                            .with_props(file_io_props)
+                            .build()?;
+                        (location, file_io)
+                    }
+                };
+
+                // Create the full path to table metadata by combining the object store location and
+                // relative table metadata path
+                let absolute_path = format!(
+                    "{}/{}",
+                    location.trim_end_matches("/"),
+                    cmd.path.trim_start_matches("/")
+                );
+                let iceberg_table = StaticTable::from_metadata_file(
+                    &absolute_path,
+                    TableIdent::from_strs(vec![DEFAULT_SCHEMA, "dummy_name"]).unwrap(),
+                    file_io,
+                )
+                .await?
+                .into_table();
```

Comment:
I think this (or the line below) will only work if the table already exists at the passed location, whereas we support creating a brand new table from scratch for Delta.

Maybe `LakehouseSyncTarget::Iceberg` should only keep hold of `FileIO`, and then in `flush_syncs` we check whether a table exists, and if not create one first similar to here https://github.com/splitgraph/seafowl/blob/54bc3046e7f728a1fa9a30a025022c1ee7d6ee79/src/sync/writer.rs#L414-L422